inductiva/tasks/task.py

```diff
@@ -408,3 +410,58 @@ def get_machine_type(self) -> Optional[str]:
         machine_type = machine_info["vm_type"].split("/")[-1]
 
         return machine_type
+
+    def get_stdout(self, n_lines: int = 10):
+        """Returns tail of stdout.txt file for current task
+
+        Calls the get_file_tail function, specifying the path of the
+        stdout_live.txt file, that contains the live version of the
+        stdout.txt file. The function returns the last n_lines in a list
+        of lines.
+
+        Args:
+            n_lines: Number of lines to return from the end of the file.
+        Returns:
+            A list of strings, each string being a line from the stdout."""
+
+        #This path needs to be updated with the self.parent_dir
+        # when that feature is added to tasks
+        path = os.path.join(self.id, "stdout_live.txt")
```

Comment:
The code is updated with the new feature that Cristiana implemented.
I understand your concerns, but the implementation you suggest requires the stdout file to be tracked at all times, even in non related functions like the ones that change the name of the files.
Since this function is to be mainly used as the simulation runs, to check its status, I don't think that we need to adress that issue. Wdyt?

Reply:
Tracked as in you need to register somewhere what is the "stdout" file that you want to display when you call get_stdout in the client. That would require checks to the function change_filename, that checks if we are changing the name / moving the stdout file

Reply:
That approach (get the stdout in the backend by specifying the path there) was the initial draft of this method, but @hpenedones pointed out that, for the sake of modularity, we should have the generic tail endpoint and pass the specific path of stdout.txt later.

If I pass it to the backend the changes in future modifications of the way we read stdout will be the same regardless of where I have the function (change the name for example).

Reply:
I think what @luispcunha is suggesting, which makes sense: is that the client should not need to be aware of those filenames that are stored in the backend (which could potentially be changed in the future).

You can have the get stdout of a task implemented in the backend, while still respecting what I suggested: implement first a function in the backend that can head or tail the content of any file (done!), and then have an backend endpoint for a task stdout calls that function. In the client package you can just call that backend endpoint directly. 

In other words: I basically agree with everything @luispcunha said in its first comment in this discussion (what you did is "almost there", you just need to move this to the backend and in the client call it with one line)

Reply:
> is that the client should not need to be aware of those filenames that are stored in the backend
But the client is not aware of that, since it only calls get_stdout(), and never specifies the path.

What I don't understand is the need of having three function (standard tail -> tail for the stdout -> get_stdout in the client) instead of the cleaner two (standard tail -> get_stdout in the client)

Reply:
Ok I understand the logic. I though that the way it is implemented is cleaner, since it has only two dependencies (tail endpoint -> get_stdout method) instead of three (tail endpoint -> tail endpoint for stdout -> get_stdout method), but the points raised by Luis also make sense

Reply:
Makes perfect sense! Thanks